index.js

```diff
@@ -281,6 +281,32 @@ function parse(input, options) {
 exports.extract = extract;
 exports.parse = parse;
 
+function pick(input, keysToPick) {
+	const {url, query} = parseUrl(input);
+	const search = extract(input);
+	const getParameterOrder = key => search.indexOf(`${key}=`) === 0 ? search.indexOf(`${key}=`) : search.indexOf(`&${key}=`) === -1 ? -1 : search.indexOf(`&${key}=`);
+	const sortedKeys = Object.keys(query).sort((keyA, keyB) => getParameterOrder(keyA) - getParameterOrder(keyB));
+	const filteredParameters = sortedKeys.filter(key => keysToPick.includes(key)).map(key => `${key}=${query[key]}`);
+	const queryString = search && filteredParameters.length > 0 ? `?${filteredParameters.join('&')}` : '';
+	const hash = getHash(input);
+	return `${url}${queryString}${hash}`;
+}
```

Comment:
I think the logic here could be simplified by making use of the existing methods in this package.

Reply:
I agree, but couldn't find out how. Any idea for the one I could use? Maybe using an intermediate function that assembles `url` `queryString` and `hash` and that I would use in both `stringifyUrl` and `pick`?
The thing is, I can't find another function that'd preserve the parameters order :/

Reply:
Yes, try that. It would also nice to DRY up the repeated code in `pick` and `exclude`.